Classes/Domain/Model/Frontend/OpeningHours.php

```diff
@@ -48,10 +50,26 @@ class OpeningHours implements TypeInterface, \Iterator, \Countable
     public function __construct(string $serialized)
     {
         $this->serialized = $serialized;
-        $this->array = array_map(
+        $this->array = $this->createArray($serialized);
+    }
+
+    private function createArray(string $serialized): array
+    {
+        $array = array_map(
             [OpeningHour::class, 'createFromArray'],
             json_decode($serialized, true) ?? []
         );
+
+        $today = GeneralUtility::makeInstance(Context::class)->getPropertyFromAspect('date', 'full');
```

Comment:
I don't like the code as it is duplicated from Import/Entity and uses makeInstance().